src/writer/blocks/sharedblueprint.py

```diff
@@ -0,0 +1,98 @@
+"""
+Shared blueprint implementation for user-created reusable blueprints.
+
+Shared blueprints execute stored blueprint components by loading them from the component tree.
+The component's content.sourceBlueprintId field references the source blueprint.
+"""
+
+from typing import TYPE_CHECKING, Dict
+
+import writer.core_ui
+from writer.abstract import register_abstract_template
+from writer.blocks.base_block import BlueprintBlock
+from writer.ss_types import AbstractTemplate
+
+if TYPE_CHECKING:
+    from writer.blueprints import BlueprintRunner
+
+
+class SharedBlueprint(BlueprintBlock):
+    """
+    Shared blueprint that executes components from a source blueprint in the component tree.
+
+    The source blueprint is identified by the sourceBlueprintId content field.
+    """
+
+    @classmethod
+    def register(cls, type: str):
+        """Register the shared_blueprint type with its abstract template."""
+        super(SharedBlueprint, cls).register(type)
+        register_abstract_template(
+            type,
+            AbstractTemplate(
+                baseType="blueprints_node",
+                writer={
+                    "name": "Shared Blueprint",
+                    "description": "Executes a shared blueprint from the component tree.",
+                    "category": "Logic",
+                    "toolkit": "blueprints",
+                    "fields": {
+                        "payload": {
+                            "name": "Payload",
+                            "desc": "The value specified will be available using the template syntax, e.g. @{payload}.",
+                            "default": "{}",
+                            "type": "Text",
+                            "control": "Textarea",
+                        },
+                    },
+                    "outs": {
+                        "success": {
+                            "name": "Success",
+                            "description": "Blueprint executed successfully.",
+                            "style": "success",
+                        },
+                        "error": {
+                            "name": "Error",
+                            "description": "Blueprint execution failed.",
+                            "style": "error",
+                        },
+                    },
+                },
+            ),
+        )
+
+    def __init__(
+        self,
+        component: writer.core_ui.Component,
+        runner: "BlueprintRunner",
+        execution_environment: Dict,
+    ):
+        super().__init__(component, runner, execution_environment)
+        self.source_blueprint_id = component.content.get("sourceBlueprintId", "")
+
+    def run(self):
+        """Execute the source blueprint's components."""
+        if not self.source_blueprint_id:
+            raise ValueError("No source blueprint ID specified")
+
+        try:
+            payload = self._get_field("payload")
```

Comment:
If you use object payload

```suggestion
            payload = self._get_field("payload", True)
```